src/fromager/sdist.py

```diff
@@ -126,6 +126,9 @@ def handle_requirement(
         servers = [sources.PYPI_SERVER_URL]
         if ctx.wheel_server_url:
             servers.insert(0, ctx.wheel_server_url)
+        if pbi.wheel_server_url:
+            # use only the wheel server from settings if it is defined. Do not fallback to other URLs
+            servers = [pbi.wheel_server_url]
```

Comment:
It's a little subtle to notice that line 131 replaces the variable and that's the intent. Something like this emphasizes that more:

```suggestion
        if pbi.wheel_server_url:
            # use only the wheel server from settings if it is defined. Do not fallback to other URLs
            servers = [pbi.wheel_server_url]
        else:
            servers = [sources.PYPI_SERVER_URL]
            if ctx.wheel_server_url:
                servers.insert(0, ctx.wheel_server_url)
```